### PR TITLE
Pawsey recipe healpix cxx

### PR DESCRIPTION
**Pull request by pelahi**
Also update to healpy

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-healpy/package.py [hunk 1]
```diff
@@ -25,3 +25,5 @@ class PyHealpy(PythonPackage):
     depends_on("py-six", type=("build", "run"))
     depends_on("cfitsio", type=("build", "run"))
     depends_on("healpix-cxx", type=("build", "run"))
+
+    patch("cfitsio_version_check.patch", when="@1.13:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-healpy/package.py:29]: This patch doesn't exist. Can you also add a comment explaining why it's needed?

Comment (pelahi) [reply var/spack/repos/builtin/packages/py-healpy/package.py:29]: The patch is necessary for the code to compile. The original code made some assumptions when checking the version of a dependency that does not hold true. The patch is not fully fleshed out as it is not clear what the developers actually need to check when compiling the code. Definitely, the current check is inadequate and will fail with newer `cfitsio` versions. The patch effectively removes the check and I rely on spack to determine whether the satisfies the version requirements.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-healpy/package.py:29]: Is it the same patch for both healpix-cxx and py-healpy? If so you'll need to copy it to the py-healpy directory and/or modify it.

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while, and tests are not passing. Feel free to reopen if you want to continue work on this.

**Comment by pelahi** [comment]
I will revisit this to see how newer versions of the software packages behave. Perhaps it is fixed.

**Pull request closed without merging**